rclrs/src/node/mod.rs

```diff
@@ -40,6 +40,28 @@ pub struct Node {
     pub(crate) subscriptions: Vec<Weak<dyn SubscriptionBase>>,
 }
 
+/// A builder for node instantiation.
+///
+/// default value in each fields are
+///   - context: user defined
+///   - name: user defined
+///   - namespace: "/"
+///
+/// # Example
+/// ```
+/// # use rclrs::{Context, Node, NodeBuilder, RclrsError};
+/// let context = Context::new([])?;
+/// let node = NodeBuilder::new("foo_node", &context).namespace("/bar").build()?;
+/// assert_eq!(node.name(), "foo_node");
+/// assert_eq!(node.namespace(), "/bar");
+/// # Ok::<(), RclrsError>(())
+/// ```
+pub struct NodeBuilder {
+    context: Arc<Mutex<rcl_context_t>>,
+    name: String,
```

Comment:
We could make `name` and `namespace` public, that way not only chaining style is supported, but also assignment style.

Reply:
Well, I don't think encapsulation matters much here: Everybody already knows that `name` and `namespace` are part of the builder, and there is no invariant of `NodeBuilder` that needs to be ensured by the public API (every `String` is valid here since validation only happens in `build()`).

But I'm also fine with leaving it this way.